server/controllers/users/changePassword.js

```diff
@@ -1,3 +1,23 @@
-exports.changePassword = (req, res) => {
-  res.send('hello from changePassword file');
+const { hash, compare } = require('bcrypt');
+const { users: { changePassword, getUserById } } = require('../../database/queries');
+
+exports.changePassword = (req, res, next) => {
+  const { userId, oldPassword, newPassword } = req.body;
+  getUserById(userId).then(({ rows }) => {
+    if (rows.length === 0) throw new Error();
+    return Promise.all([
+      compare(oldPassword, (rows[0].password)),
+      compare(newPassword, rows[0].password),
+    ]);
+  }).then(([oldPasswordMatches, newPasswordMatches]) => {
+    if (!oldPasswordMatches) throw new Error('You entered a wrong password');
+    else if (newPasswordMatches) throw new Error('Your new password and old password are the same');
+    return hash(newPassword, 10);
+  })
+    .then((hashedPassword) => changePassword(userId, hashedPassword))
+    .then(() => res.send('Password is changed'))
```

Comment:
If we send data we agreed to send it in a JSON format and it's a common practice to also add the status code if you want, just a suggestion ;)
```js
res.status(200).json({ status: 200, message: 'Password is changed' });
```

Reply:
good comment @yosefalnajjarofficial  
but why you sent the status code with the response :
```js
   res.status(200).json({ message: 'Password is changed' });

```